BuildAndTest.cmd

```diff
@@ -1,116 +1,125 @@
 @ECHO off
-SETLOCAL
-@REM Uncomment this line to update nuget.exe
-@REM Doing so can break SLN build (which uses nuget.exe to
-@REM create a nuget package for the SARIF SDK) so must opt-in
-@REM %~dp0.nuget\NuGet.exe update -self
+SETLOCAL ENABLEDELAYEDEXPANSION
+
+rem Uncomment this line to update nuget.exe
+rem Doing so can break SLN build (which uses nuget.exe to
+rem create a nuget package for the SARIF SDK) so must opt-in
+rem %~dp0.nuget\NuGet.exe update -self
 
-set Platform=AnyCPU
 set Configuration=Release
+set SolutionFile=src\Everything.sln
+set FullClean=false
+set NoClean=false
+set NoBuild=false
+set NoTest=false
+set NoPublish=false
+set ThisFileDir=%~dp0
 
 :NextArg
 if "%1" == "" goto :EndArgs
 if "%1" == "/config" (
     if not "%2" == "Debug" if not "%2" == "Release" echo error: /config must be either Debug or Release && goto :ExitFailed
     set Configuration=%2&& shift && shift && goto :NextArg
 )
+if "%1" == "/sln" (
+    if "%2" == "" echo error: no argument specified for /sln && goto ExitFailed
+    if not exist "%2" echo error: solution file "%2" does not exist && goto ExitFailed
+    set SolutionFile=%2&& shift && shift && goto :NextArg
+)
+if "%1" == "/fullclean" (
+    set FullClean=true&& shift && goto :NextArg
+)
+if "%1" == "/noclean" (
+    set NoClean=true&& shift && goto :NextArg
+)
+if "%1" == "/nobuild" (
+    REM If we're not going to build, we'd better not clean.
+    set NoBuild=true&& set NoClean=true&& shift && goto :NextArg
+)
+if "%1" == "/notest" (
+    set NoTest=true&& shift && goto :NextArg
+)
+if "%1" == "/nopublish" (
```

Comment:
does this mean the script by default tries to publish? that seems wrong. this should not be the default. I'll look elsewhere to confirm

Reply:
Yes, but "publish" doesn't mean what you think :-) The `dotnet publish` command just creates a directory that includes all of the program's runtime dependencies, so you can xcopy deploy it. The "publish" portion of BuildAndTest just runs that program to create the directory, for example `bld\bin\Sarif.Multitool\AnyCPU_Release\Publish\net461\`.